tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -770,9 +769,29 @@ def with_airbyte_python_connector_full_dagger(context: ConnectorContext, build_p
     )
 
 
-def with_crane(context: PipelineContext) -> Container:
+def with_crane(
+    context: PipelineContext,
+) -> Container:
     """Crane is a tool to analyze and manipulate container images.
     We can use it to extract the image manifest and the list of layers or list the existing tags on an image repository.
     https://github.com/google/go-containerregistry/tree/main/cmd/crane
     """
-    return context.dagger_client.container().from_("gcr.io/go-containerregistry/crane:v0.15.1")
+
+    # We use the debug image as it contains a shell which we need to properly use environment variables
+    # https://github.com/google/go-containerregistry/tree/main/cmd/crane#images
+    base_container = context.dagger_client.container().from_("gcr.io/go-containerregistry/crane/debug:v0.15.1")
+
+    if context.docker_hub_username_secret and context.docker_hub_password_secret:
+        base_container = (
+            base_container.with_secret_variable("DOCKER_HUB_USERNAME", context.docker_hub_username_secret).with_secret_variable(
+                "DOCKER_HUB_PASSWORD", context.docker_hub_password_secret
+            )
+            # We need to use skip_entrypoint=True to avoid the entrypoint to be overridden by the crane command
+            # We use sh -c to be able to use environment variables in the command
+            # This is a workaround as the default crane entrypoint doesn't support environment variables
+            .with_exec(
+                ["sh", "-c", "crane auth login index.docker.io -u $DOCKER_HUB_USERNAME -p $DOCKER_HUB_PASSWORD"], skip_entrypoint=True
```

Comment:
@alafanechere 

I went on way to long of a deep dive here but essentially I think this is our best way to do this.

Here are a few roadblocks I found
1. cranes default entrypoint didnt support env vars in any of the following formats `$ENV` `${ENV}` `"$ENV"` `\"$ENV\"`
2. crane did not have a shell in the image by default, you need to use a `debug` image to get access
3. I couldnt use a dagger secret inline `with_exec("auth", "login", "index.docker.io", "-u", context.docker_hub_username_secret)`

The only alternative I found was explictly reading the secrets and passing them in as plain text

```python
    async def _run(self) -> StepResult:
        docker_repository, docker_tag = self.context.docker_image_name.split(":")
        dhu = await self.context.docker_hub_username_secret.plaintext()
        dhs = await self.context.docker_hub_password_secret.plaintext()
        crane_ls = (
            environments.with_crane(
                self.context,
                dhu,
                dhs,
            )
```

Which I think is worse.

Let me know your thoughts